### PR TITLE
Change pull request template

### PR DESCRIPTION
**Pull request by tbsiqueira**
## Problem
In general pull requests are merged without a clear definition of done. This causes some issues that could be avoided by using a simple list to guide developers on how to consider a task really done.

## Solution
Improve the pull request template and add an extra **_Defintion of done section_**, also improve the same template to add more explained data

## Issue tracker
N/A

## How to test
N/A

## Screenshots
N/A

## Release notes
N/A

## Change Record
N/A

## Translations
N/A

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,29 +1,49 @@
...
 - [ ] Using version X.Y.Z of Open Social with the example module enabled
 - [ ] As a sitemanager
 - [ ] Try to enable the option B on screen c/d/e
 - [ ] When saving I expect the result to be F but instead see G.
 - [ ] The expected result F is attained when repeating the steps with this fix applied.
 
+## Definition of done
+*[Required]*
+- [ ] Code/peer review is completed
+- [ ] All commit messages are clear and clean. If applicable a rebase was performed
```
Comment (ronaldtebrake) [inline .github/PULL_REQUEST_TEMPLATE.md:24]: Would be great to link to a piece of documentation here, it's a bit ambigious what clear and clean mean for us while we do have some requirements. Maybe we can make something public in Slite for this?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,29 +1,49 @@
...
+- [ ] All automated tests are green
+- [ ] Functional/manual tests of the acceptance criteria are approved
+- [ ] All acceptance criteria were meet
```
Comment (ronaldtebrake) [inline .github/PULL_REQUEST_TEMPLATE.md:27]:
```suggestion
- [ ] All acceptance criteria were met
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,29 +1,49 @@
...
+- [ ] If applicable (I.E. new hook_updates) the update path from previous versions (major and minor versions) are tested
+- [ ] Code is tested on all branches that it has been cherry-picked
```
Comment (ronaldtebrake) [inline .github/PULL_REQUEST_TEMPLATE.md:29]: We might need to split this up before / after merge. 
Testing this on all cherry-picked branches can only be done when cherry-picked and only if applicable. 
Update hook numbers also might need to be adjusted on cherry-pick so we can also include that.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,29 +1,49 @@
...
+- [ ] This pull request has all required labels (team/type/priority)
+- [ ] This pull request has a milestone
+- [ ] This pull request has an assignee ( if applicable)
```
Comment (ronaldtebrake) [inline .github/PULL_REQUEST_TEMPLATE.md:32]:
```suggestion
- [ ] This pull request has an assignee (if applicable)
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -1,29 +1,49 @@
...
+- [ ] The Drupal.org ticket is updated according to this pull reflex status
```
Comment (ronaldtebrake) [inline .github/PULL_REQUEST_TEMPLATE.md:33]:
```suggestion
- [ ] The Drupal.org ticket is updated according to this pull request status
```

Comment (ronaldtebrake) [inline .github/PULL_REQUEST_TEMPLATE.md:33]:
```suggestion
- [ ] The Drupal.org ticket(s) are updated according to this pull request status
```

**Comment by tbsiqueira** [comment]
CHeery-picked to 11.0.x and 11.1.x

**Pull request merged**